Core.DomainServices/Generic/IEntityIdentityResolver.cs

```diff
@@ -12,5 +12,7 @@ public interface IEntityIdentityResolver
     {
         Maybe<Guid> ResolveUuid<T>(int dbId) where T : class, IHasUuid, IHasId;
         Maybe<int> ResolveDbId<T>(Guid uuid) where T : class, IHasUuid, IHasId;
+        Result<int, OperationError> ExchangeId<TEntity>(Guid uuid) where TEntity : Entity, IHasUuid;
```

Comment:
we talked about extension methods... this is changing the interface. Please add as extebsuib netgids in stead. That keeps the complexity down of this class as our new methods are just a specific use case of the resolver.